ticketbuyer/purchase.go

```diff
@@ -65,4 +74,25 @@ out:
 			break out
 		}
 	}
+	p.wg.Done()
+}
+
+// Start starts the purchase manager goroutines.
+func (p *PurchaseManager) Start() {
+	p.wg.Add(1)
+	go p.NotificationHandler()
+}
+
+// WaitForShutdown blocks until all purchase manager goroutines have finished executing.
+func (p *PurchaseManager) WaitForShutdown() {
```

Comment:
Yeah, we want to wait on this before calling `UnloadWallet`. Problem is the instance of `PurchaseManager` changes depending on the chain client connection, so need some way to always wait on the latest instance.